Custom Field Constraints/CFConstraints.state.interrupter.slice.js

```diff
@@ -19,28 +19,68 @@ tau.mashups
                 return this.sliceDecoder.decode(change.id);
             },
 
-            _getNewState: function(entity, entityStatesDetailed, changesToInterrupt, defaultProcess) {
+            _getNewState: function(entity, entityStatesDetailed, changesToInterrupt, defaultProcess, teamProjects) {
                 var entityStateChange = _.find(changesToInterrupt, function(change) {
-                    return parseInt(this.sliceDecoder.decode(change.id)) == entity.id;
+                    return parseInt(this.sliceDecoder.decode(change.id)) === entity.id;
                 }, this);
 
-                return this._getEntityState(entity, entityStatesDetailed, entityStateChange.changes, defaultProcess);
+                return this._getEntityState(entity, entityStatesDetailed, entityStateChange.changes, defaultProcess, teamProjects);
             },
 
-            _getEntityState: function(entity, entityStates, changes, defaultProcess) {
-                var stateName = this.sliceDecoder.decode(_.find(changes,function(change) {
+            _getEntityState: function(entity, entityStates, changes, defaultProcess, teamProjects) {
+                var change = _.find(changes, function(change) {
                     return this._shouldChangeBeHandled(change);
-                }, this).value);
+                }, this);
+                var stateName = this.sliceDecoder.decode(change.value);

```

Comment:
If `_.find()` won't find any change that should be handled, `change.value` throws.